lib/tree.rb

```diff
@@ -16,42 +16,100 @@ def initialize
     @root = nil
   end
 
-  # Time Complexity: 
-  # Space Complexity: 
+  # Time Complexity: O(log n) 
+  # Space Complexity: O(1)
```

Comment:
Since you are doing recursion here, this is O(log n) assuming the tree is balanced.  Remember the system stack used by recursion.